TLM/TLM/UI/ModUI.cs

```diff
@@ -235,8 +228,9 @@ public class UIOpacityObservable : GenericObservable<UIOpacityNotification> {
             Log._Debug("ModUI.OnLevelLoaded: called");
             if (ModUI.Instance == null) {
```

Comment:
`ModUI.Instance` is based on `UICustomControl` which is based on `MonoBehaviour` which does freaky shit with `null`.

This should be changed to `if (ModUI.Instance) ...` ?

Reply:
Didn't make any changes here. It was not necessary for the task at hand (removal of Finalizers). So if it worked before, it should work afterwards.

Reply:
It is likely not working though, because anything based on `MonoBehaviour` is doing weird shit with `null`.

@krzychu124 any comment?

Reply:
It should be ok if we nullify `Instance` just after `Destroy()`

Reply:
Behaviour is similar although not the same as before.

Before:
LoadingExtension did Object.Destroy(ModUI.Instance), then used the static method SetSingletonInstance to set function to null.
Then GC then called cleanup stuff implicitly by calling the Finalizer.

After:
Since I removed the Finalizer, I made the Destroy method instead for the cleanup stuff.
And I call that from LoadingExtension.
That method does explicitly what was done before implicetly, then sets its own Instance (now only a private setter) to null, then Destroys itself.

The steps are the same, the order is reversed though.